Jenkinsfile

```diff
@@ -0,0 +1,79 @@
+pipeline {
+
+    agent { label "codejail-worker" }
+
+    options {
+        timeout(30)
+    }
+    stages {
+        stage('Install tox') {
+            steps {
+                sh '''
+                virtualenv tox_env;
+                source tox_env/bin/activate;
```

Comment:
Would the `withPythonEnv` step from [pyenv-pipeline](https://plugins.jenkins.io/pyenv-pipeline) be a little cleaner here?  I'm using that in the devstack snapshot builder.

Reply:
Done